internal/model/resource.go

```diff
@@ -0,0 +1,18 @@
+package model
+
+import (
+	"time"
+
+	"github.com/shopspring/decimal"
+)
+
+type Resource struct {
+	ID        int
+	CreatedAt time.Time
+	UpdatedAt time.Time
+	DeletedAt *time.Time
+
+	Name        string
+	Amount      decimal.Decimal
```

Comment:
Aqui pode ser um `float32`

```suggestion
	Amount      float32
```